presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -426,7 +463,26 @@ public TimeBoundary(
 
     public TimeBoundary getTimeBoundaryForTable(String table)
     {
-        String responseBody = sendHttpGetToBroker(table, String.format(TIME_BOUNDARY_API_TEMPLATE, table));
-        return timeBoundaryJsonCodec.fromJson(responseBody);
+        try {
+            String responseBody = sendHttpGetToBroker(table, String.format(TIME_BOUNDARY_API_TEMPLATE, table));
+            return timeBoundaryJsonCodec.fromJson(responseBody);
+        }
+        catch (Exception e) {
+            if ((e instanceof PinotException)) {
+                /** New Pinot broker will set response code 404 if time boundary is not set.
+                 * This is backward incompatible with old version, as it returns an empty json object.
+                 * In order to gracefully handle this, below check will extract response code and return empty json
+                 * if not found time boundary.
+                 *
+                 * Sample error message:
+                 *     Unexpected response status: 404 for request  to url http://127.0.0.1:8000/debug/timeBoundary/baseballStats, with headers ...
+                 */
+                String[] errMsgSplits = e.getMessage().split(" ");
```

Comment:
s/errMsgSplits/errorMessageSplits/g
minor thing, Presto usually not use abbreviation in variable name

Reply:
Sure! Still getting used to Presto coding style :)